core/src/main/kotlin/no/strooped/StroopedController.kt

```diff
@@ -5,30 +5,39 @@ import com.badlogic.gdx.Screen
 import com.badlogic.gdx.utils.ObjectMap
 import no.strooped.service.GameLifecycleService
 import no.strooped.service.JoinGameService
+import no.strooped.service.SocketService
+import no.strooped.view.screen.EndRoundScreen
 import no.strooped.view.screen.JoinGameScreen
 import no.strooped.view.screen.LobbyScreen
+import no.strooped.view.screen.TaskScreen
+import no.strooped.view.screen.components.ColorButton
 
 const val TITLE = "Strooped"
 
 class StroopedController : Game() {
     private val screens: ObjectMap<Class<out Screen>, Screen> = ObjectMap()
-    var joinGameService: JoinGameService = JoinGameService(socketService = Any())
-    var gameLifecycleService: GameLifecycleService = GameLifecycleService()
+    var socket: SocketService = SocketService()
+    var joinGameService: JoinGameService = JoinGameService(socket)
+    var gameLifecycleService: GameLifecycleService = GameLifecycleService(socket)
 
     override fun create() {
         loadScreens()
         inititalizeServices()
-        changeScreen(JoinGameScreen::class.java)
+        screens.put(EndRoundScreen::class.java, EndRoundScreen(this))
+        changeScreen(EndRoundScreen::class.java)
         gameLifecycleService.onNextTask {
-            // screens.put(TaskScreen::class.java, TaskScreen(this, GameSingleton.room!!))
-            // changeScreen(TaskScreen::class.java)
+            GameSingleton.taskNumber++
+            screens.put(TaskScreen::class.java, TaskScreen(this, it))
+            changeScreen(TaskScreen::class.java)
             // add task to singleton
             // showNextTask(task)
         }
-        gameLifecycleService.onTaskTimeout {
-            // disable answering
-        }
-        gameLifecycleService.onRoundEnd {
+        gameLifecycleService.onRoundEnd { // task is called round:ending..will receive a player object
+            GameSingleton.room?.player?.let {
```

Comment:
I can recommend using `apply` instead of `let` here. It lets you modify the object inside the player context, so you don't need to use `it`

```kotlin
GameSingleton.room?.player?.apply {
  placement = 1
}
```

Reply:
done